### PR TITLE
Disable usage of library ShortcutBadger on device running API 26+

### PR DESCRIPTION
**Pull request by bmarty**

### CODE REVIEW

File: vector/src/main/java/im/vector/ui/badge/BadgeProxy.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ */
+object BadgeProxy {
+    private const val LOG_TAG = "BadgeProxy"
+
+    /**
+     * Badge is now managed by notification channel, so no need to use compatibility library in recent versions
+     *
+     * @return true if library ShortcutBadger can be used
+     */
+    private fun useShortcutBadger(): Boolean {
```
Comment (BillCarsonFr) [inline vector/src/main/java/im/vector/ui/badge/BadgeProxy.kt]: use short syntax

File: vector/src/main/java/im/vector/ui/badge/BadgeProxy.kt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        return Build.VERSION.SDK_INT < Build.VERSION_CODES.O
+    }
+
+    /**
+     * Update the application badge value.
+     *
+     * @param context    the context
+     * @param badgeValue the new badge value
+     */
+    fun updateBadgeCount(context: Context, badgeValue: Int) {
```
Comment (BillCarsonFr) [inline vector/src/main/java/im/vector/ui/badge/BadgeProxy.kt]: I think we might need to update notification channels default settings to state if we want to display or not badges (setShowBadge(true/false)).
And will probably need some work on notification builder with .setNumber(messageCount)

Comment (bmarty) [reply vector/src/main/java/im/vector/ui/badge/BadgeProxy.kt]: Yes

File: vector/src/main/java/im/vector/ui/badge/BadgeProxy.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/ui/badge/BadgeProxy.kt]: Done

**Pull request merged**